nav2_collision_monitor/src/polygon.cpp

```diff
@@ -54,21 +134,98 @@ int Polygon::getPointsInside(const std::vector<Point> & points)
   return num;
 }
 
-bool Polygon::getParameters(std::string & polygon_topic) {
+void Polygon::publish(const std::string & base_frame_id) const
+{
+  if (!visualize_) {
+    return;
+  }
+
+  auto node = node_.lock();
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node"};
+  }
+
+  // Fill PolygonStamped struct
+  std::unique_ptr<geometry_msgs::msg::PolygonStamped> poly_s =
+    std::make_unique<geometry_msgs::msg::PolygonStamped>();
+  poly_s->header.stamp = node->now();
+  poly_s->header.frame_id = base_frame_id;
+  poly_s->polygon = polygon_;
+
+  // Publish polygon
+  polygon_pub_->publish(std::move(poly_s));
+}
+
+bool Polygon::getBasicParameters(std::string & polygon_topic)
+{
   auto node = node_.lock();
   if (!node) {
     throw std::runtime_error{"Failed to lock node"};
   }
 
-  PolygonBase::getParameters(polygon_topic);  // Will always return true
+  // Get action type
+  nav2_util::declare_parameter_if_not_declared(
+    node, polygon_name_ + ".action_type", rclcpp::ParameterValue("stop"));  // Stop by default
```

Comment:
There should be no default

Reply:
[Changed](https://github.com/ros-planning/navigation2/pull/2982/files#diff-e7fd1328f82b3191cac8b30062c6eee0ea702a0be381dde8572120fa21a87595R229)